lib/locale_threads.t

```diff
@@ -4,27 +4,441 @@ use warnings;
+                            som         => 'arabic',
+                            spa         => 'latin_1',
+                            sr          => 'cyrillic',
+                            srp         => 'cyrillic',
+                            tam         => 'tamil',
+                            ta          => 'tamil',
+                            tg          => 'cyrillic',
+                            tgk         => 'cyrillic',
+                            tha         => 'thai',
+                            th          => 'thai',
+                            thai        => 'thai',
+                            ti          => 'ethiopian',
+                            tir         => 'ethiopian',
+                            uk          => 'cyrillic',
+                            ukr         => 'cyrillic',
+                            ur          => 'arabic',
+                            urd         => 'arabic',
+                            zgh         => 'arabic',
+                            zh          => 'chinese',
+                            zho         => 'chinese',
+                        );
+my %codeset_to_script = (
+                            88591  => 'latin_1',
+                            88592  => 'latin_2',
+                            88593  => 'latin_3',
+                            88594  => 'latin_4',
+                            88595  => 'cyrillic',
+                            88596  => 'arabic',
+                            88597  => 'greek',
+                            88598  => 'hebrew',
+                            88599  => 'latin_5',
+                            885910 => 'latin_6',
+                            885911 => 'thai',
+                            885912 => 'devanagari',
+                            885913 => 'latin_7',
+                            885914 => 'latin_8',
+                            885915 => 'latin_9',
+                            885916 => 'latin_10',
+                            cp1251 => 'cyrillic',
+                            cp1255 => 'hebrew',
+                      );
+
+my %script_priorities = (       # In trying to make the results as distinct as
+                                # possible, make the ones closest to Unicode,
+                                # and ASCII lowest priority
+                            $official_ascii_name => 15,
+                            latin_1 => 14,
+                            latin_9 => 13,
+                            latin_2 => 12,
+                            latin_4 => 12,
+                            latin_5 => 12,
+                            latin_6 => 12,
+                            latin_7 => 12,
+                            latin_8 => 12,
+                            latin_10 => 12,
+                            latin   => 11,  # Unknown latin version
+                        );
+
+my %script_instances;   # Keys are scripts, values are how many locales use
+                        # this script.
+
+sub analyze_locale_name($) {
```

Comment:
While not an issue for this commit, I wonder how much of this could go in a module.

Reply:
I don't know of other uses for this.